<!--

Brings up a version of move_base that runs in the odom_combined frame, and 
doesn't use a map.  It runs in the move_base_local namespace so it can run
at the same time as the standard nav stack, and in fact assumes that the
sensor processing, etc., has been brought up by wg_pr2_2dnav/pr2_2dnav.launch
or equivalent.

-->

<launch>
  <include file="$(find pr2_machine)/$(env ROBOT).machine" />
  <node pkg="flirtlib_ros" name="startup_loc_wander" type="wander"
        required="true" output="screen"/>
  <node ns="move_base_local" pkg="move_base" type="move_base" name="move_base_node" machine="c2">
    <remap from="odom" to="/base_odometry/odom" />
    <remap from="cmd_vel" to="/navigation/cmd_vel" />

    <!-- Load common configuration files -->
    <rosparam file="$(find flirtlib_ros)/launch/local_nav/move_base_params.yaml" command="load"/>
    <rosparam file="$(find pr2_navigation_config)/move_base/costmap_common_params.yaml"
              command="load" ns="global_costmap" />
    <rosparam file="$(find pr2_navigation_config)/move_base/costmap_common_params.yaml"
              command="load" ns="local_costmap" />
    <rosparam file="$(find flirtlib_ros)/launch/local_nav/base_local_planner_params.yaml"
              command="load" />
    <rosparam file="$(find flirtlib_ros)/launch/local_nav/local_costmap_params.yaml"
              command="load" ns="local_costmap"/>
    <rosparam file="$(find flirtlib_ros)/launch/local_nav/global_costmap_params.yaml"
              command="load" ns="global_costmap"/>
  </node>
</launch>
